tests/core/mempool/test_mempool_manager.py

```diff
@@ -136,7 +178,92 @@ async def test_block_cost_exceeds_max() -> None:
 async def test_double_spend_prevalidation() -> None:
     mempool_manager = await instantiate_mempool_manager(zero_calls_get_coin_record)
     conditions = [[ConditionOpcode.CREATE_COIN, IDENTITY_PUZZLE_HASH, 1]]
-    sb = spend_bundle_from_conditions(conditions)
+    sb = spend_bundle_from_conditions(TEST_COIN, conditions)
     sb_twice: SpendBundle = SpendBundle.aggregate([sb, sb])
     with pytest.raises(ValidationError, match="Err.DOUBLE_SPEND"):
         await mempool_manager.pre_validate_spendbundle(sb_twice, None, sb_twice.name())
+
+
+def test_run_for_cost_and_additions() -> None:
+    conditions = [[ConditionOpcode.CREATE_COIN, IDENTITY_PUZZLE_HASH, 1]]
+    solution = Program.to(conditions)
+    with pytest.raises(ValueError, match="('cost exceeded', '01')"):
+        run_for_cost_and_additions(TEST_COIN_ID, IDENTITY_PUZZLE, solution, uint64(1))
+    created_coin = Coin(TEST_COIN_ID, IDENTITY_PUZZLE_HASH, 1)
+    expected_cost, expected_coins = run_for_cost_and_additions(TEST_COIN_ID, IDENTITY_PUZZLE, solution, uint64(50))
+    assert expected_cost == 44
+    assert expected_coins == [created_coin]
+
+
+@pytest.mark.asyncio
+async def test_check_item_for_dedup() -> None:
+    async def get_coin_record(coin_id: bytes32) -> Optional[CoinRecord]:
+        test_coin_records = {
+            TEST_COIN_ID: TEST_COIN_RECORD,
+            TEST_COIN_ID2: TEST_COIN_RECORD2,
+            TEST_COIN_ID3: TEST_COIN_RECORD3,
+        }
+        return test_coin_records.get(coin_id)
+
+    mempool_manager = await instantiate_mempool_manager(get_coin_record)
+
+    # No eligible coins, nothing to deduplicate, item gets considered normally
+    conditions = [[ConditionOpcode.AGG_SIG_UNSAFE, G1Element(), IDENTITY_PUZZLE_HASH]]
+    _, sb_name, res = await generate_and_add_spendbundle(mempool_manager, TEST_COIN, conditions)
+    assert res == (2873056, MempoolInclusionStatus.SUCCESS, None)
+    mempool_item = get_mempool_item(mempool_manager, sb_name)
+    dedup_coin_spends: Dict[bytes32, DedupCoinSpend] = {}
+    result = check_item_for_dedup(mempool_item, dedup_coin_spends)
+    assert result == ([], 0, {}, set())
```

Comment:
it really seems unnecessarily complicated to create a mempool and add an item to it just to test `check_item_for_dedup()`. I think you should just create the `MempoolItem` independently. It might sound like a nit, but we want to make each test case as simple and small as possible do lower the friction to add more cases.

Reply:
Done.